### PR TITLE
@metamask/eslint config@5.0.0

### PR DESCRIPTION
**Pull request by rekmarks**
It's semicolons o'clock. Recommend reviewing individual commits:

- 21d7f7b: rc-file changes
- 68962f3: `yarn lint:fix` (don't bother with this)
- d5ad26e: package update
- d5717e1: let `prettier` handle semicolons

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -147,7 +140,9 @@ module.exports = {
     'no-invalid-this': 'off',
     '@babel/no-invalid-this': 'error',
 
-    '@babel/semi': ['error', 'never'],
+    semi: 'off',
+    '@babel/semi': 'error',
```
Comment (brad-decker) [inline .eslintrc.js:144]: I recommend turning this off... prettier will throw errors for issues with semicolons and will automatically fix with --fix

Comment (rekmarks) [reply .eslintrc.js:144]: Hm, do you know if `prettier` handle class properties? See `@babel/semi` comment in readme here: https://github.com/babel/babel/tree/main/eslint/babel-eslint-plugin

**Comment by kumavis** [comment]
why tho
i dissent

**Comment by rekmarks** [comment]
@kumavis most of the discussion for why is on Slack, I believe. A plurality of North America TZ extension devs were positive, some neutral, and I think no one outright against it.

File: .eslintrc.js (hunk 1 above)
Comment (brad-decker) [reply .eslintrc.js:144]: it's turned off in eslint-config-prettier, so i'd assume so. https://github.com/prettier/eslint-config-prettier/blob/main/babel.js

Comment (rekmarks) [reply .eslintrc.js]: Done in: d5717e1

**Review by brad-decker — APPROVED** [review]

**Pull request merged**